pkg/proxy/ipvs/proxier.go

```diff
@@ -1619,7 +1619,8 @@ func (proxier *Proxier) cleanLegacyService(activeServices map[string]bool, curre
 
 func (proxier *Proxier) cleanLegacyBindAddr(activeBindAddrs map[string]bool, currentBindAddrs []string) {
 	for _, addr := range currentBindAddrs {
-		if _, ok := activeBindAddrs[addr]; !ok {
+		// Strip IP mask from currentBindAddr (e.g. 1.2.3.4/32) to match activeBindAddrs (e.g. 1.2.3.4)
+		if _, ok := activeBindAddrs[strings.Split(addr, "/")[0]]; !ok {
```

Comment:
How can we confirm that the addr is in the form of `1.2.3.4/32` instead of `1.2.3.4`?

What will happen here if the addr is `1.2.3.4` without the `/32`?

Reply:
kube-proxy should stop passing around string addresses and instead pass around `net.IP` and `net.IPNet` (after all that's what they're for and it'd solve many of the IPv4/IPv6 problems stemming from throwing away info). However, that'd be a larger refactor.

This fix tries to make do with the existing strings. The split is written to defensively allow either `1.2.3.4` or `1.2.3.4/32` rather than crashing. https://play.golang.org/p/BswmkSLBELv